src/librustc_metadata/diagnostics.rs

```diff
@@ -26,6 +26,28 @@ name. Example:
 ```
 "##,
 
+E0455: r##"
+A framework link kind was used and the OS used to compile isn't the OS X
```

Comment:
Perhaps:

"Linking with `kind=framework` is only supported when targeting OS X, as frameworks are specific to that operating system."?